dashboard/app/controllers/api/v1/pd/workshops_controller.rb

```diff
@@ -15,7 +15,18 @@ def index
       @workshops = @workshops.organized_by(current_user)
     end
 
-    render json: @workshops, each_serializer: Api::V1::Pd::WorkshopSerializer
+    if params[:include_enrollments]
+      enrollments = Pd::Enrollment.where(email: current_user.email)
+      return_val = []
+
+      @workshops.each do |workshop|
+        return_val << Api::V1::Pd::WorkshopSerializer.new(workshop, scope: enrollments.find_by(pd_workshop_id: workshop.id)).attributes
```

Comment:
It's the same either way. How about doing one more scoped initial query like:

```ruby
enrollments = Pd::Enrollment.where(pd_workshop_id: @workshops.pluck(:id)).where('email = ? OR user_id = ?', current_user.email, current_user.id)
```

You might also need to do `.all` or `.to_a` to make the query one time instead of every time you subsequently filter. Play around with it.

Or you could do a similar thing that puts them into a hash by workshop id.